src/main/java/net/snowflake/ingest/streaming/internal/MemoryInfoProviderFromRuntime.java

```diff
@@ -4,20 +4,44 @@
 
 package net.snowflake.ingest.streaming.internal;
 
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
 /** Reads memory information from JVM runtime */
 public class MemoryInfoProviderFromRuntime implements MemoryInfoProvider {
-  @Override
-  public long getMaxMemory() {
-    return Runtime.getRuntime().maxMemory();
+  private final long maxMemory;
+  private volatile long totalFreeMemory;
+  private final ScheduledExecutorService executorService;
+
+  public MemoryInfoProviderFromRuntime(long freeMemoryUpdateIntervalMs) {
+    maxMemory = Runtime.getRuntime().maxMemory();
```

Comment:
How is `maxMemory` calculated? Can this change over time or is this fixed at JVM start?

Reply:
IIUC, this comes from the `-Xmx` parameter of the JVM and therefore, cannot change.